### ISSUE TITLE
"Dafny > test"

### ISSUE THREAD

**Issue by MikaelMayer** [dafny-lang/ide-vscode#417]
I would like to just right-click and run Dafny > test, but this option is not present.
Currently, I select "build with custom arguments" and I replace everything with "test" or "test --no-verify".
it would be good to have the two options.
Test without verification is useful to check up if a property holds before proving it.

**Comment by keyboardDrummer** [comment]
Related/duplicate: https://github.com/dafny-lang/ide-vscode/issues/418

**Comment by MikaelMayer** [comment]
Ok thanks for the clarification. From the documentation you shared with me, it looks like designing a user-level task generator is a bit more ambitious than what I implemented. It would be certainly be helpful for a class of Dafny users who otherwise just switch to the command line and write their various "make xxx" commands because they usually put everything in a makefile. Being able to get these commands from the dafny project sounds a good idea too.

In my case and with a lot of new Dafny users, they don't necessarily have project files,  everything they need at the beginning is a way to run and test their files, using a keyboard shortcut (F5 or SHIFT+F5 as I suggest). It would be a bit overwhelming to have to define them the notion of project file/task/generate task/modify task.

So in short, I like your solution in general, but for what our need is currently, I wouldn't spend time myself implementing it.

**Comment by keyboardDrummer** [comment]
> From the documentation you shared with me, it looks like designing a user-level task generator is a bit more ambitious than what I implemented. 

Does it not seem about equivalent in simplicity? You define objects like the JSON I quoted, but then in code. You don't have to do anything dynamic or do anything with project files.

**Comment by MikaelMayer** [comment]
> Does it not seem about equivalent in complexity? You define objects like the JSON I quoted, but then in code. You don't have to do anything dynamic or do anything with project files.

For now it looks like it's more obvious to you than to me. What I did was straightforward just to unblock new and existing users discovering the Run function and in need of the test function. I am still not sure about the tasks because I haven't looked deeply at them. I'd be happy to review a proposal about tasks, or even just a comparative workflow.

> I would be careful in adding these commands, if later we might decide to use tasks instead and then want to remove the commands.

I think that, for discoverability purposes, I would very likely vote against any removal of commands in general easily accessible from the right-click, unless there is another compelling UI we can start using (e.g. the Run and Debug interface). Again, I have not studied tasks closely so I can't say if they will fit our needs or not, and from what I heard of you, I'm not yet convinced (but happy to change mind with more evidence).

**Comment by MikaelMayer** [comment]
I should recall about at least one user who would probably like to configure tasks if ever you want to make someone happier.

https://github.com/dafny-lang/ide-vscode/issues/383
